mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -178,19 +196,8 @@ public String apply(String input) {
     }
 
     private Optional<Cid> cid(Optional<ContentIdField> contentIdField) {
-        return contentIdField.transform(new Function<ContentIdField, Optional<Cid>>() {
-            @Override
-            public Optional<Cid> apply(ContentIdField field) {
-                return Optional.fromNullable(field.getId())
-                        .transform(new Function<String, Cid>() {
-
-                            @Override
-                            public Cid apply(String cid) {
-                                return Cid.from(cid);
-                            }
-                        });
-            }
-        }).or(Optional.<Cid> absent());
+        return contentIdField.transform(CONTENT_ID_FIELD_TO_OPTIONAL_CID_FUNCTION)
+                .or(Optional.<Cid> absent());
```

Comment:
Let see the type of "contentIdField" is Optional. This case mean
- "contentIdField" should transform to Optional<Cid> if it present
- .or if "contentIdField" is absent.
Is that correct?